### PR TITLE
[bug/enhancement] Multiple fixes / extra formatting

### PR DESCRIPTION
**Pull request by TJulesL**
Added :

- Font loading, code now can load fonts from a string. In old code it wasn't clear how to load fonts or if you could load them at all. Also added an check that when an error occurs it prints it and uses the system default font

- Fixed borders for circles / rectangles, It didnt work at all at first because there was no logic implemented. I have implemented it so it draws for example a big circle (border) with a little circle within (inner circle).

- play.new_text now gives an specific error when the input is not a string

- fixed images, you are now able to load images in with a location on the computer as string 

- Added some extra "documentation" in generator.py

- Added a new feature so you can smooth out the corners with a given "border_radius". I will implement functionality later so it also works for rectangles without any borders, since this is not yet possible at the moment

### CODE REVIEW

**Comment by koen1711** [comment]
Hi thank you so much for your suggestions and fixes, we are currently heavily working on the documentation pages and adding more tests etc. Let me know if you find anymore bugs so those can be addressed

**Comment by koen1711** [comment]
I'll review all of this tonight and then let you know what I think, in my big refactor one or two of the logic might have became unimplemented which in the future won't happen again of course. Since that would be devastating for the classroom

**Comment by TJulesL** [comment]
Is goed neem je tijd, als project ga ik nu ook onder begeleiding van meneer postma werken (als het goed is) dus ik zal waarschijnlijk nog wel vaker een paar pull requests of eventuele features zal committen (achtergrondmuziek / sound effects implementeren etc).

File: play/api/generators.py
Comment (koen1711) [inline play/api/generators.py:16]: We use None for a reason here to load, the default font. Since not everyone uses windows I would recommend using a more general file path like /path/to/file

File: play/api/generators.py
Comment (koen1711) [inline play/api/generators.py:22]: -> _Text

File: play/api/generators.py
Comment (koen1711) [inline play/api/generators.py:63]: _Box

File: play/api/generators.py
Comment (koen1711) [inline play/api/generators.py:103]: _Circle

File: play/api/generators.py
Comment (koen1711) [inline play/api/generators.py:140]: _Line

File: play/api/generators.py
Comment (koen1711) [inline play/api/generators.py:175]: _Image

**Comment by koen1711** [comment]
Please also fix your code using `black ./play ./tests` and `pylint ./play`

**Comment by koen1711** [comment]
Also, before merging. Please change the version in setup.py using semantic versioning

**Comment by TJulesL** [comment]
I have changed the version, a minor version increment because of the added features/fixes. Also changed the text.py file to now use logging.py. At last i also used black / pylint to format it

**Comment by TJulesL** [comment]
i just noticed a fatal error in the code which glitches out the objects (ball and box) so **Please dont merge just yet**. I will try and fix it right now

**Comment by TJulesL** [comment]
i just noticed a fatal error in the code which glitches out the objects (ball and box) so **Please dont merge just yet**. I will try and fix it right now

**Comment by TJulesL** [comment]
> > i just noticed a fatal error in the code which glitches out the objects (ball and box) so **Please dont merge just yet**. I will try and fix it right now
> 
> Alright, could you show the code where that happens? Because if you update box.y it's logical it doesn't work since you do it yourself.



So the problem wasn't my code which i thought it would be (something like the borders i implemented), it was that there was no code for drawing the rotation of boxes and circles. This made it so that pymunk calculated the collision correctly however the box was never drawn on that angle. The reason why it looked like a piece of the box was "under the screen" was because the x/y coords for the middle were calculated with the context of the rectangle laying on its side. This also explains the weird collision in parts where the box wasn't. 

I coded the logic for it and fixed it + added it in one of my recent commits where i formatted it with black and pylint. Also tested it again, collision with different types of objects work fine now + the rounding of borders etc.

**Review by koen1711 — APPROVED** [review]

**Pull request merged**
